public/app/features/dimensions/editors/ResourcePicker.tsx

```diff
@@ -0,0 +1,139 @@
+    if (currentFolder) {
+      getBackendSrv()
+        .get(`public/${currentFolder?.value}/index.json`)
+        .then((data) => {
+          const cards = data.files.map((v: string) => ({
+            value: v,
+            label: v,
+            imgUrl: `public/${currentFolder?.value}/${v}`,
+          }));
+          setDirectoryIndex(cards);
+          setDefaultList(cards);
+        })
+        .catch((e) => console.error(e));
+    } else {
+      return;
+    }
+  }, [currentFolder]);
+
+  const onChangeSearch = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.value) {
+      const filtered = directoryIndex.filter((card) =>
+        card.value
+          // exclude file type (.svg) in the search
+          .substr(0, card.value.length - 4)
+          .toLocaleLowerCase()
+          .includes(e.target.value.toLocaleLowerCase())
+      );
+      setDirectoryIndex(filtered);
+    } else {
+      setDirectoryIndex(defaultList);
+    }
+  };
+  const imgSrc = value.indexOf(':/') > 0 ? value : 'public/' + value;
+
+  return (
+    <div>
+      <Label>Current Item</Label>
+      <div style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center', columnGap: '2px' }}>
+        {mediaType === 'icon' && <SVG src={imgSrc} width="40" height="40" fill={theme.colors.text.primary} />}
+        {mediaType === 'image' && <img src={imgSrc} width="40" height="40" />}
+        <StringValueEditor value={value} onChange={onChange} item={{} as any} context={{} as any} />
+        <Button variant="secondary" onClick={() => onChange(value)}>
+          Apply
+        </Button>
+      </div>
+      <TabsBar>
+        {tabs.map((tab, index) => (
+          <Tab
+            label={tab.label}
+            key={index}
+            active={tab.active}
+            onChangeTab={() => setTabs(tabs.map((tab, idx) => ({ ...tab, active: idx === index })))}
+          />
+        ))}
+      </TabsBar>
+      <TabContent>
+        {tabs[0].active && (
+          <div className={styles.tabContent}>
+            <Select options={folders} onChange={setCurrentFolder} value={currentFolder} />
+            <Input placeholder="Search" onChange={onChangeSearch} />
+            {directoryIndex ? (
+              <div className={styles.cardsWrapper}>
+                <ResourceCards cards={directoryIndex} onChange={onChange} currentFolder={currentFolder} />
+              </div>
+            ) : (
+              <Spinner />
+            )}
+          </div>
+        )}
+        {/* TODO: add file upload
+          {tabs[1].active && (
+          <FileUpload
+            onFileUpload={({ currentTarget }) => console.log('file', currentTarget?.files && currentTarget.files[0])}
+            className={styles.tabContent}
+          />
+        )} */}
+      </TabContent>
+    </div>
+  );
+}
+
+const getStyles = stylesFactory((theme: GrafanaTheme2, color) => {
+  return {
+    cardsWrapper: css`
+      height: calc(100vh - 490px);
```

Comment:
my attempt to avoid double scrollbar... but still :cry: 
![image](https://user-images.githubusercontent.com/705951/132044184-0bb8f826-bd29-440f-879c-7744b10f99a6.png)